src/sentry/static/sentry/app/components/letar.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+const Letar = React.createClass({
+  propTypes: {
+    user: React.PropTypes.object.isRequired
+  },
+
+  COLORS: {
+    blue: '#25A6F7',
+    green: '#6FBA57',
+    orange: '#EC5B33',
+    red: '#E35141',
+    pink: '#E56AA6',
+    purple: '#836CC2'
+  },
+
+  getColor() {
+    // If you change anything with color selection,
+    // make sure you also change it in avatar.py
+    let color = 'purple';
+    let userId = this.getIdentifierHash();
+    if (userId % 6 === 0) {
+      color = 'blue';
+    } else if (userId % 5 === 0) {
+      color = 'green';
+    } else if (userId % 4 === 0) {
+      color = 'orange';
+    } else if (userId % 3 === 0) {
+      color = 'red';
+    } else if (userId % 2 === 0) {
+      color = 'pink';
+    }
+    return this.COLORS[color];
+  },
+
+  hashIdentifier(identifier) {
+    let hash = 0;
+    for (let i = 0; i < identifier.length; i++) {
+      hash += identifier.charCodeAt(i);
+    }
+    return hash;
+  },
+
+  getIdentifierHash() {
+    let user = this.props.user;
+    let identifier = user.email || user.username || user.id || user.ip_address;
+    return this.hashIdentifier(identifier);
+  },
+
+  getDisplayName() {
+    let user = this.props.user;
+    return user.name || user.email || user.username || '?';
+  },
+
+  getInitials() {
+    let names = this.getDisplayName().split(' ');
+    let initials = names[0][0] + (names.length > 1 ? names[names.length - 1][0] : '');
+    return initials.toUpperCase();
+  },
+
+  render() {
+    return (
+      <svg viewBox="0 0 120 120" xmlns="http://www.w3.org/2000/svg" className={this.props.className}>
```

Comment:
Since this is duplicated inside Python as well, this should be commented in both places to suggest changing one requires changing the other so these don't diverge.